vm/process/systemVM.go

```diff
@@ -129,6 +129,17 @@ func (s *systemVM) RunSmartContractCall(input *vmcommon.ContractCallInput) (*vmc
 	return vmOutput, nil
 }
 
+// GasScheduleChange sets the new gas schedule where it is needed
+func (s *systemVM) GasScheduleChange(gasSchedule map[string]map[string]uint64) {
+	apiCosts := gasSchedule[core.ElrondAPICost]
+	if apiCosts == nil {
+		return
+	}
+
+	s.asyncCallStepCost = apiCosts[core.AsyncCallStepField]
```

Comment:
We need to protect these with a mutex

Reply:
no need. there is no possibility to call the from different go routines

Reply:
protected all the way